### ISSUE TITLE
NaN in /federate endpoint when timeseries disappears

### ISSUE THREAD

**Issue by afoninsky** [VictoriaMetrics/VictoriaMetrics#3185]
**Describe the bug**
In our current architecture we have a set of vmsignle instances, and global prometheus. Prometheus pulls metrics from instances using [federation](https://prometheus.io/docs/prometheus/latest/federation/. When timeseries disappears in vmsingle - it is still visible for some time in global prometheus as NaN. I.e. /federate endpoint exposes non-existing data (most likely cause, not checked)

The reason it is considered as a bug - the same workflow works fine when prometheus instead of vmsingle.

**Blast radius**
It breaks all rules/queries evaluation as most of math functions return NaN when any argument is NaN.

**Expected behavior**
Missing timeseries is not delivered via federation as NaN

**Screenshots**
vmsingle non-existing metric:
![2022-09-30 09 20 42](https://user-images.githubusercontent.com/15193941/193213981-4c249a85-08bd-40e3-b4aa-231f9607bdfe.jpg)

result in prometheus collected using /federate:
![2022-09-30 09 20 49](https://user-images.githubusercontent.com/15193941/193214081-97d08951-f65b-4553-b0e2-636ba51899ed.jpg)

**Version**
```
Prometheus 2.17.2
victoria-metrics-20220908-105456-tags-v1.81.2-0-gdd1126dcc
```

**Used command-line flags**
```
/bin/vmsingle -storageDataPath=/var/lib/victoria-metrics-data -retentionPeriod=7d -httpListenAddr=127.0.0.1:9091 -maxLabelsPerTimeseries=40 -maxInsertRequestSize=1024MB -search.maxConcurrentRequests=22 -search.maxQueueDuration=60s -search.cacheTimestampOffset=6m -search.maxFederateSeries=5000000 -search.maxUniqueTimeseries=1000000 -search.maxQueryDuration=60s -memory.allowedPercent=80 -enableTCP6=true -promscrape.maxScrapeSize=1024MB -promscrape.minResponseSizeForStreamParse=512MB -promscrape.config.strictParse=false -promscrape.streamParse=true -promscrape.suppressDuplicateScrapeTargetErrors=true -promscrape.config=/etc/prometheus/prometheus.yaml -vmalert.proxyURL=http://127.0.0.1:8880
```

**Comment by valyala** [comment]
It has been appeared that VictoriaMetrics could return `NaN` values for stale metrics at [/federate](https://docs.victoriametrics.com/#federation) endpoint. This has been fixed in the commit fcc7ab71b3daba61f518c61ba0cf129b752c35bd - now VictoriaMetrics no longer export stale metrics at `/federate` endpoint.

This commit will be included in the next release. In the mean time it is possible to test the bugfix by building VictoriaMetrics from the given commit according to [these docs](https://docs.victoriametrics.com/#how-to-build-from-sources).

**Comment by valyala** [comment]
FYI, the bugfix has been included into [VictoriaMetrics v1.79.4 LTS release](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v1.79.4).

**Comment by valyala** [comment]
VictoriaMetrics shouldn't expose NaN values for stale series at [/federate](https://docs.victoriametrics.com/#federation) endpoint starting from [v1.82.0](https://github.com/VictoriaMetrics/VictoriaMetrics/releases).

Closing the issue as fixed.

@afoninsky , feel free re-opening the issue if it still reproduces in v1.82.0 or later releases.
